core/docker/build.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF 1>&2
+Usage: $0 [-v <VERSION>] [-h]
+Builds the Trino Docker image
+
+-h       Display help
+-v       Build the specified Trino version, downloads all required artifacts
+EOF
+}
+
+TRINO_VERSION=
+
+while getopts ":h:v:" o; do
+    case "${o}" in
+        v)
+            TRINO_VERSION=${OPTARG}
+            ;;
+        h)
+            usage
+            exit 0
+            ;;
+        *)
+            usage
+            exit 1
+            ;;
+    esac
+done
+shift $((OPTIND - 1))
+
+SOURCE_DIR="../.."
+WORK_DIR="$(mktemp -d)"
+
+# Retrieve the script directory.
+SCRIPT_DIR="${BASH_SOURCE%/*}"
+cd "${SCRIPT_DIR}" || exit 2
+
+if [ -n "$TRINO_VERSION" ]; then
+    SERVER_LOCATION="https://repo1.maven.org/maven2/io/trino/trino-server/${TRINO_VERSION}/trino-server-${TRINO_VERSION}.tar.gz"
+    CLIENT_LOCATION="https://repo1.maven.org/maven2/io/trino/trino-cli/${TRINO_VERSION}/trino-cli-${TRINO_VERSION}-executable.jar"
+    curl -o "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz" "${SERVER_LOCATION}"
+    curl -o "${WORK_DIR}/trino-cli-${TRINO_VERSION}-executable.jar" "${CLIENT_LOCATION}"
+    chmod +x "${WORK_DIR}/trino-cli-${TRINO_VERSION}-executable.jar"
+else
+    # Move to the root directory to run maven for current version.
+    pushd ${SOURCE_DIR}
+    TRINO_VERSION=$(./mvnw --quiet help:evaluate -Dexpression=project.version -DforceStdout)
+    popd
+
+    cp "${SOURCE_DIR}/core/trino-server/target/trino-server-${TRINO_VERSION}.tar.gz" "${WORK_DIR}"
+    cp "${SOURCE_DIR}/client/trino-cli/target/trino-cli-${TRINO_VERSION}-executable.jar" "${WORK_DIR}"
+fi
+
+tar -C "${WORK_DIR}" -xzf "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz"
+rm "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz"
+cp -R bin "${WORK_DIR}/trino-server-${TRINO_VERSION}"
+cp -R default "${WORK_DIR}/"
+
+CONTAINER="trino:${TRINO_VERSION}"
+
+docker build "${WORK_DIR}" --pull --platform linux/amd64 -f amd64.dockerfile -t "${CONTAINER}-amd64" --build-arg "TRINO_VERSION=${TRINO_VERSION}"
+docker build "${WORK_DIR}" --pull --platform linux/arm64 -f arm64.dockerfile -t "${CONTAINER}-arm64" --build-arg "TRINO_VERSION=${TRINO_VERSION}"
+
+rm -r "${WORK_DIR}"
+
+# Source common testing functions
+# shellcheck source=core/docker/container-test.sh
```

Comment:
Can this be a relative path? It's in the same directory as this script, so it seems unfortunate we need an absolute path here

Reply:
TIL that if I configure my editor to run shellcheck with `-P SCRIPTDIR` it's not necessary at all, so I removed it.